packages/kbn-config-schema/src/internals/index.ts

```diff
@@ -274,6 +274,54 @@ export const internals = Joi.extend([
       },
     ],
   },
+  {
+    name: 'record',
+    pre(value: any, state: State, options: ValidationOptions) {
+      if (!isPlainObject(value)) {
+        return this.createError('record.base', { value }, state, options);
+      }
+
+      return value as any;
+    },
+    rules: [
+      anyCustomRule,
+      {
+        name: 'entries',
+        params: { key: Joi.object().schema(), value: Joi.object().schema() },
```

Comment:
I'm confused, here we define params as an object with key and value properties, but when people call `Joi.record().entries()` the key and value params are passed as positional arguments. Is an options object supported too? Does the order of keys in the params object define the validation order for arguments?

Reply:
I know it's super weird, but their alternative with `paramN` is not much better IMO, but tell me what you prefer and I'll use that: https://github.com/hapijs/joi/blob/master/API.md#extension
> The params of rules rely on the fact that all engines, even though not stated in the ECMA specifications, preserve the order of object keys, this is a conscious choice to simplify the API for the end-user. If you ever see an engine misbehaving or are uncomfortable relying on this, you can use a single option object to describe your parameters, like:
> params: { options: Joi.object({ param1: Joi.number().required(), param2: Joi.string() }) }

Reply:
Alright, that's fine since it's the API setup by Joi, but yeah, not a fan of that design.